cwms-data-api/src/main/java/cwms/cda/data/dao/watersupply/WaterSupplyUtils.java

```diff
@@ -158,4 +168,251 @@ public static WATER_USER_CONTRACT_TAB_T toWaterUserContractTs(WaterUserContract
+        String pumpLocation = acctObjT.getPUMP_LOCATION_REF().call_GET_LOCATION_ID();
+        String pumpOffice = acctObjT.getPUMP_LOCATION_REF().getOFFICE_ID();
+        PumpTransfer transfer = null;
+        if (waterUserContract.getPumpInLocation().getPumpLocation().getName().equalsIgnoreCase(pumpLocation)
+                && waterUserContract.getPumpInLocation().getPumpLocation().getOfficeId().equalsIgnoreCase(pumpOffice)) {
+            transfer = new PumpTransfer(PumpType.IN, acctObjT.getPHYSICAL_TRANSFER_TYPE().getDISPLAY_VALUE(),
+                    acctObjT.getPUMP_FLOW(), acctObjT.getACCOUNTING_REMARKS());
+        } else if (waterUserContract.getPumpOutLocation().getPumpLocation().getName().equalsIgnoreCase(pumpLocation)
+                && waterUserContract.getPumpOutLocation().getPumpLocation().getOfficeId().equalsIgnoreCase(pumpOffice)) {
+            transfer = new PumpTransfer(PumpType.OUT, acctObjT.getPHYSICAL_TRANSFER_TYPE().getDISPLAY_VALUE(),
+                    acctObjT.getPUMP_FLOW(), acctObjT.getACCOUNTING_REMARKS());
+        } else if (waterUserContract.getPumpOutBelowLocation().getPumpLocation().getName().equalsIgnoreCase(pumpLocation)
+                && waterUserContract.getPumpOutBelowLocation().getPumpLocation().getOfficeId().equalsIgnoreCase(pumpOffice)) {
+            transfer = new PumpTransfer(PumpType.BELOW, acctObjT.getPHYSICAL_TRANSFER_TYPE().getDISPLAY_VALUE(),
+                    acctObjT.getPUMP_FLOW(), acctObjT.getACCOUNTING_REMARKS());
+        }
+        if (transfer != null) {
+            pumpAccounting.put(acctObjT.getTRANSFER_START_DATETIME().toInstant(), Collections.singletonList(transfer));
+        }
+        return new WaterSupplyAccounting.Builder()
+            .withContractName(acctObjT.getWATER_USER_CONTRACT_REF().getCONTRACT_NAME())
+            .withWaterUser(toWaterUser(waterUserObjT))
+            .withPumpLocations(new PumpLocation.Builder()
+                .withPumpIn(CwmsId.buildCwmsId(waterUserContract.getPumpInLocation().getPumpLocation().getOfficeId(),
+                    waterUserContract.getPumpInLocation().getPumpLocation().getName()))
+                .withPumpOut(CwmsId.buildCwmsId(waterUserContract.getPumpOutLocation().getPumpLocation().getOfficeId(),
+                    waterUserContract.getPumpOutLocation().getPumpLocation().getName()))
+                .withPumpBelow(CwmsId.buildCwmsId(waterUserContract.getPumpOutBelowLocation().getPumpLocation().getOfficeId(),
+                            waterUserContract.getPumpOutBelowLocation().getPumpLocation().getName()))
+                .build())
+            .withPumpAccounting(pumpAccounting)
+            .build();
+    }
+
+    private static void addTransfer(WAT_USR_CONTRACT_ACCT_OBJ_T acctObjTs, WaterSupplyAccounting accounting) {
+        PumpTransfer transfer = null;
+        if (accounting.getPumpLocations().getPumpIn().getName().equalsIgnoreCase(acctObjTs.getPUMP_LOCATION_REF().call_GET_LOCATION_ID())
```

Comment:
`accounting.getPumpLocations().getPumpIn()` etc can be null

Reply:
Added null handling